storage/storage.cpp

```diff
@@ -1461,25 +1494,8 @@ void Storage::OnDiffStatusReceived(diffs::Status const status)
     {
       auto const countryId = FindCountryIdByFile(localDiff.GetCountryName());
 
-      auto const registerDiffFn = [this, countryId](bool isSuccess)
-      {
-        ASSERT_THREAD_CHECKER(m_threadChecker, ());
-        if (!isSuccess)
-        {
-          m_failedCountries.insert(countryId);
-          NotifyStatusChangedForHierarchy(countryId);
-          return;
-        }
-
-        TLocalFilePtr localFile = GetLocalFile(countryId, GetCurrentDataVersion());
-        ASSERT(localFile, ());
-        DeleteCountryIndexes(*localFile);
-        m_didDownload(countryId, localFile);
-        NotifyStatusChangedForHierarchy(countryId);
-      };
-
       if (m_diffManager.HasDiffFor(countryId))
-        ApplyDiff(countryId, registerDiffFn);
+        UpdateNode(countryId);
```

Comment:
Could you please comment what was wrong with the old code here?

Reply:
This was a situation when we found a valid diff and started to apply it (applying happens on the diff manager worker thread). At the same time, a user could initiate updating the same mwm by using the downloader. So at the point when applying is finished (after calling of ApplyDiff here), the downloader could initiate new applying because it had just downloaded the same diff file. But an old mwm which is necessary for applying was already deleted in the previous applying so it was CHECK for this situation.
With new approach, user will just see a regular situation as he could see before he closed the app: the mwm is being updated and it can't be canceled because it's already in applying state and in another hand the user can't start new updating/downloading of the same mwm, because it's being updated right now.